### PR TITLE
Remove black as a dependency 

### LINKED ISSUE #34: Strict dependency on black prevents usage of the latest black
`pipenv-setup` depends on `black==19.10b0`, while the latest black version is `20.8b1`. This prevents the usage of `pipenv-setup` in any project which requires the latest black version (downgrading `black` is not an option, because `19.10b0` complains about formatting of the files `20.8b1` is happy with).

### PR DESCRIPTION
**Pull request by benhowes**
Fixes #34

Originally this PR just attempted to relax the version pinning, however, that did not work as expected.

Looking at the code in more detail it seems that this library includes support for not having black present in any case, so it's probably better to not have it as a hard dependency.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -143,11 +143,11 @@
             "pytest-xdist~=1.29",
             "tox~=3.14",
             "autopep8~=1.4",
+            "black~=20.8b1; python_version >= '3.6'",
```
Comment (benhowes) [inline setup.py:146]: I assumed that an update to the latest version would not be a problem!

Comment (grizz) [reply setup.py:146]: black formats differently based on its version, so black should probably be pinned in the projects using this library, not here. If it must be here, then probably having it be any version makes more sense?

Comment (grizz) [reply setup.py:146]: @Madoshakalaka thoughts? This issue (#34) is causing problems for us.

Comment (benhowes) [reply setup.py:146]: @grizz I've moved the requirement to the dev requirements in this PR, so it won't attempt to install black at all when you use it.

Comment (bryant-finney) [reply setup.py:146]: I'm also running into this issue. Wouldn't upgrading the `dev` black dependency require reformatting this code base (though I suppose a follow-up PR would be appropriate for that)? 

Aside from that, I def like this solution, FWIW.

File: pipenv_setup/setup_updater.py [hunk 2]
```diff
@@ -147,18 +147,21 @@ def format_file(file):  # type: (Path) -> None
...
-
-        code = autopep8.fix_code(file.read_text())
-        file.write_text(code)
+        try:
+            import autopep8
+        else:
+            code = autopep8.fix_code(file.read_text())
+            file.write_text(code)
+        except ImportError:
+            raise RuntimeError("Either black or autopep8 must be installed.")
```
Comment (bryant-finney) [inline pipenv_setup/setup_updater.py:164]: The `else` before `except` block is a syntax error; I'd recommend the following:

```python
    except ImportError:
        # use autopep8
        try:
            # third party
            import autopep8
        except ImportError:
            raise RuntimeError("Either black or autopep8 must be installed.")
        else:
            code = autopep8.fix_code(file.read_text())
            file.write_text(code)
    else:
        with Popen(
            [sys.executable, "-m", "black", str(file)], stdout=PIPE, stderr=PIPE
        ) as p:
            p.communicate()
```

**Comment by benhowes** [comment]
Solved better in https://github.com/Madoshakalaka/pipenv-setup/pull/88

**Pull request closed without merging**